client/context/query.go

```diff
@@ -313,3 +386,21 @@ func (ctx CLIContext) queryStore(key cmn.HexBytes, storeName, endPath string) ([
 	path := fmt.Sprintf("/store/%s/%s", storeName, endPath)
 	return ctx.query(path, key)
 }
+
+// isQueryStoreWithProof expects a format like /<queryType>/<storeName>/<subpath>
+// queryType can be app or store
+// if subpath equals to store or key, then return true
+func isQueryStoreWithProof(path string) (bool) {
+	if !strings.HasPrefix(path, "/") {
+		return false
+	}
+	paths := strings.SplitN(path[1:], "/", 3)
+	if len(paths) != 3 {
+		return false
+	}
+	// WARNING This should be consistent with query method in iavlstore.go
```

Comment:
Can you elaborate on this warning and why we need it? What needs to be done about it?

Reply:
In [iavlstore.go](https://github.com/cosmos/cosmos-sdk/blob/24c7cb03d3ec1dd5a9ec2f0215c358db5af70191/store/iavlstore.go#L212), proof will be included in response only when the query path is /store or /key. I concern that this may be changed in iavlstore.go. If there are some changes about proof building in iavlstore.go, then we must change code here to be consistent with the changes.